### ISSUE TITLE
Address DAP violations for v9

### ISSUE THREAD

**Issue by astha-jain** [carbon-design-system/carbon#3481]
## DAP violations for List Box fix needed for v9

DAP violations for List Box (#2827) were fixed for V10 in https://github.com/carbon-design-system/carbon/pull/3139. This fix should be available for v9 as well. 

## Environment
See #2827

## Detailed description
See #2827

## Steps to reproduce the issue
See #2827

@emyarod

**Comment by emyarod** [comment]
@astha-jain you can test the fix here https://github.com/carbon-design-system/carbon-components-react/pull/2400

**Comment by xubinnb** [comment]
Hi @emyarod , when will we have the same fix for v9?

**Comment by emyarod** [comment]
@xubinnb this should be in the latest v9 release https://github.com/carbon-design-system/carbon-components-react/releases/tag/v6.115.6

**Comment by astha-jain** [comment]
@emyarod This did not fix the DAP violation. DAP violation is `Accessible name does not contain the visible label text`. That said, aria label is not same as visible label. This got me investigating. 
ListBox or ListBox Field are not getting proper aria labels when used from DropdownV2 component. 
ListBox Field is a `role=button` that makes it mandatory to have an aria label but I don't see it being passed from anywhere. Could you check this out please?


<img width="698" alt="Screen Shot 2019-08-07 at 5 20 18 PM" src="https://user-images.githubusercontent.com/21059894/62659168-a049a980-b938-11e9-995e-f5fb4a0e3845.png">
<img width="696" alt="Screen Shot 2019-08-07 at 5 22 14 PM" src="https://user-images.githubusercontent.com/21059894/62659169-a049a980-b938-11e9-9ecd-849bef0a8a3c.png">


Notice aria label passed down for ListBox and not for ListBox Field.
https://github.com/carbon-design-system/carbon-components-react/blob/v6/src/components/DropdownV2/DropdownV2.js#L223

Aria label should be coming in `{...rest}` but apparently does not. 
https://github.com/carbon-design-system/carbon-components-react/blob/v6/src/components/ListBox/ListBoxField.js#L21

**Comment by emyarod** [comment]
aria label is coming in through `{...rest}` but it's being set by [downshift](https://github.com/downshift-js/downshift/tree/0319960dd76a4fbbc9d19cb7140d95db8b8106a1#getbuttonprops)

@astha-jain are you saying that `<ListBox.Field>` needs to have the same `aria-label` as `<ListBox>` or that it needs to be configurable and separate from the `aria-label` applied to `<ListBox>`?

**Comment by astha-jain** [comment]
aria label for `<ListBox.Field>` is `open menu` if you look in the first picture. It should be same as the visible label text `United States` so as to pass the accessibility. 

So this may be coming from a new WCAG 2.1 checkpoint IBM Accessibility WCAG 2.1 February 2019.

**Comment by emyarod** [comment]
opened https://github.com/carbon-design-system/carbon-components-react/pull/2410 to address this in v9
